src/util/retry.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @flow
```

Comment:
Is it worth extracting some of this repeated stuff out of XHR.js so it can be referenced in both places?

Reply:
While there is some small similarities with the retry stuff I added in xhr, it isn't really the same. This file is a simplified retry which recursively calls itself, while xhr does some specific stuff in regards to when to retry and calls axios instead of recursively calling itself

Reply:
Can we use function.js?